Stratis.Bitcoin/BlockPulling/QualityScore.cs

```diff
@@ -103,29 +92,28 @@ public void AddSample(IBlockPullerBehavior peer, long blockDownloadTimeMs, int b
 
             lock (this.lockObject)
             {
-                // If we reached the maximum number of samples, we need to remove oldest sample.
-                if (this.samplesCount == this.samples.Length)
-                {
-                    PeerSample oldSample = this.samples[this.samplesIndex];
+                // Add new sample to the mix.
+                PeerSample newSample = new PeerSample();
```

Comment:
I don't understand this part, if we call this method 1 million times wont we create 1mil new objects? 
The idea was to prevent allocation of new memory.